imagebuildah/build.go

```diff
@@ -177,7 +180,21 @@ func (b *Executor) Preserve(path string) error {
 	}
 	cacheFile := filepath.Join(cacheDir, fmt.Sprintf("volume%d.tar", b.preserved))
 	// Save info about the top level of the location that we'll be archiving.
-	archivedPath := filepath.Join(b.mountPoint, path)
+	var archivedPath string
+
+	// Try and resolve the symlink (if one exists)
+	// Set archivedPath and path based on whether a symlink is found or not
+	symLink, err := resolveSymLink(b.mountPoint, path)
+	switch err {
+	case nil:
+		archivedPath = filepath.Join(b.mountPoint, symLink)
+		path = symLink
```

Comment:
What are the benefits of changing the key that we're using for the `volumeCache` and `volumeCacheInfo` maps?

Reply:
So if after volume mounting, we decide to add a file to the path where the symlink points to fo example, the build code will fail as it won't be able to find that path in the cache maps. Changing this to be the resolved path of the symlink helps get rid of that error.